src/hooks/custom-hooks/useGetHubEnabledCountryList.ts

```diff
@@ -1,27 +1,34 @@
+import { useCallback, useEffect } from 'react';
+import { useShallow } from 'zustand/react/shallow';
 import { HUB_ENABLED_COUNTRY_LIST } from '@/constants';
+import { useHubEnabledCountryListStore } from '@/stores';
 
 /** A custom hook that returns the remote configuration to get hub_enabled_country_list. */
 const useGetHubEnabledCountryList = () => {
     const HUB_REMOTE_CONFIG_URL = process.env.VITE_HUB_REMOTE_CONFIG_URL || '';
+    const { hubEnabledCountryList, setHubEnabledCountryList } = useHubEnabledCountryListStore(
+        useShallow(state => ({
+            hubEnabledCountryList: state.hubEnabledCountryList,
+            setHubEnabledCountryList: state.setHubEnabledCountryList,
+        }))
+    );
 
-    if (HUB_REMOTE_CONFIG_URL) {
-        const getRemoteConfig = async () => {
-            const response = await fetch(HUB_REMOTE_CONFIG_URL)
-                .then(res => res.json())
-                .catch(() => {
-                    return HUB_ENABLED_COUNTRY_LIST;
-                });
+    const getRemoteConfig = useCallback(async () => {
+        try {
+            const response = await fetch(HUB_REMOTE_CONFIG_URL);
+            const data = await response.json();
+            setHubEnabledCountryList(data);
+        } catch (error) {
+            setHubEnabledCountryList(HUB_ENABLED_COUNTRY_LIST);
+        }
+    }, [hubEnabledCountryList]);
```

Comment:
Why are we creating a store for the hub enabled countries and why do we not store the value from `useGetHubEnabledCountryList()` function we call top level

Reply:
ah i see, alright 👍